UrbanEngine.Infrastructure/Repository/Repository.cs

```diff
@@ -1,83 +1,76 @@
-﻿using Microsoft.EntityFrameworkCore;
-using System;
-using System.Collections.Generic;
-using System.Linq;
+﻿namespace UrbanEngine.Infrastructure.Repository {
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Linq.Expressions;
+    using System.Threading.Tasks;
+    using Microsoft.EntityFrameworkCore;
 
-namespace UrbanEngine.Infrastructure.Repository
-{
-    public class Repository : IRepository
-    {
-        #region Fields
+    public class Repository : IRepository {
+        #region Local Members
 
-        private DbContext _dbContext = null;
+        private readonly DbContext _dbContext = null;
 
         #endregion
 
         #region Constructors
 
-        public Repository(DbContext dbContext)
-        {
+        public Repository( DbContext dbContext ) {
             _dbContext = dbContext;
         }
 
         #endregion
 
         #region Public Methods
-
-        public T GetById<T>(long id) where T : class
-        {
-            return _dbContext.Set<T>().Find(id);
+ 
+        public async Task<TEntity> SingleOrDefaultAsync<TEntity>( Expression<Func<TEntity, bool>> predicate = null ) where TEntity : class {
```

Comment:
Are we not going to have a GetById method anymore? I thought it was customary to always start with that method when building a manager class? It would be hard to do that if we don't have the method anymore.